Exception in thread "main" java.lang.NoClassDefFoundError: com/google/gson/JsonIOException

$ java -version
openjdk version "21.0.10" 2026-01-20
OpenJDK Runtime Environment (build 21.0.10+7-Ubuntu-122.04)
OpenJDK 64-Bit Server VM (build 21.0.10+7-Ubuntu-122.04, mixed mode, sharing)


$ java -jar $BIOBIN/lirical-cli-2.4.0/lirical-cli-2.4.0.jar phenopacket \
    --exomiser-hg38-dir $BIOCACHE/exomiser/2602/2602_hg38 \
    --data $BIOCACHE/lirical/data \
    --pathogenicity-threshold 0.001 \
    --threshold 0 \
    --phenopacket "${RESULTSDIR}/phenopacket.json" \
    --vcf $VCF_MERGED \
    --exomiser-hg38-clinvar $BIOCACHE/exomiser/2602/2602_hg38/2602_hg38_clinvar.mv.db \
    --exomiser-hg38-variants $BIOCACHE/exomiser/2602/2602_hg38/2602_hg38_variants.mv.db \
    --assembly hg38 \
    --transcript-db RefSeq \
    --output-format {html,tsv,json} \
    --output-directory "${RESULTSDIR}/lirical"









 _     ___ ____  ___ ____    _    _                                                       
 | |   |_ _|  _ \|_ _/ ___|  / \  | |                                                      
 | |    | || |_) || | |     / _ \ | |                                                      
 | |___ | ||  _ < | | |___ / ___ \| |___                                                   
 |_____|___|_| \_\___\____/_/   \_\_____|                                                  
                                                                                           
LIkelihood Ratio Interpretation of Clinical AbnormaLities                                  
                                    :: v2.4.0 ::                                           
                                                                                           
                                             
11:26:12.192 [main] INFO  o.m.l.c.c.LiricalConfigurationCommand - Using data folder at /mnt
/data/biocache/lirical/data
11:26:12.194 [main] INFO  o.m.l.c.c.AbstractPrioritizeCommand - Bootstrapping LIRICAL 
11:26:13.981 [main] INFO  o.m.l.c.c.AbstractPrioritizeCommand - Configured LIRICAL v2.4.0 
11:26:13.981 [main] INFO  o.m.l.c.c.AbstractPrioritizeCommand - Parsing the analysis inputs
Exception in thread "main" java.lang.NoClassDefFoundError: com/google/gson/JsonIOException
        at com.google.protobuf.util.JsonFormat$Parser.merge(JsonFormat.java:483)
        at org.phenopackets.phenopackettools.io.base.BasePhenopacketParser.readJsonMessage(
BasePhenopacketParser.java:44)
        at org.phenopackets.phenopackettools.io.base.BasePhenopacketParser.parse(BasePhenop
acketParser.java:29)
        at org.phenopackets.phenopackettools.io.PhenopacketParser.parse(PhenopacketParser.j
ava:49)
        at org.monarchinitiative.lirical.io.analysis.PhenopacketImportUtil.readPhenopacket(
PhenopacketImportUtil.java:25)
        at org.monarchinitiative.lirical.io.analysis.PhenopacketV2Importer.read(Phenopacket
V2Importer.java:30)
        at org.monarchinitiative.lirical.cli.cmd.PhenopacketUtil.readPhenopacketData(Phenop
acketUtil.java:29)
        at org.monarchinitiative.lirical.cli.cmd.PhenopacketCommand.procureSanitationInputs
(PhenopacketCommand.java:50)
        at org.monarchinitiative.lirical.cli.cmd.AbstractPrioritizeCommand.execute(Abstract
PrioritizeCommand.java:66)
        at org.monarchinitiative.lirical.cli.cmd.PhenopacketCommand.execute(PhenopacketComm
and.java:16)
        at org.monarchinitiative.lirical.cli.cmd.BaseCommand.call(BaseCommand.java:31)
        at org.monarchinitiative.lirical.cli.cmd.OutputCommand.call(OutputCommand.java:13)
        at org.monarchinitiative.lirical.cli.cmd.BaseCommand.call(BaseCommand.java:18)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(Comm
andLine.java:2469)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at org.monarchinitiative.lirical.cli.Main.main(Main.java:46)
Caused by: java.lang.ClassNotFoundException: com.google.gson.JsonIOException
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.ja
va:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders
.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
        ... 22 more

--- a/lirical-cli/pom.xml
+++ b/lirical-cli/pom.xml
@@ -91,7 +91,7 @@
                                     <goal>copy-dependencies</goal>
                                 </goals>
                                 <configuration>
-                                    <includeScope>compile</includeScope>
+                                    <includeScope>runtime</includeScope>
                                     <outputDirectory>${project.build.directory}/lib</outputDirectory>
                                     <overWriteReleases>false</overWriteReleases>
                                     <overWriteSnapshots>false</overWriteSnapshots>

--- a/lirical-io/src/test/java/org/monarchinitiative/lirical/io/analysis/PhenopacketImportUtilTest.java
+++ b/lirical-io/src/test/java/org/monarchinitiative/lirical/io/analysis/PhenopacketImportUtilTest.java
@@ -1,0 +1,24 @@
+package org.monarchinitiative.lirical.io.analysis;
+
+import org.junit.jupiter.api.Test;
+import org.phenopackets.schema.v2.Phenopacket;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class PhenopacketImportUtilTest {
+
+    @Test
+    public void readPhenopacket_protobuf() throws IOException, PhenopacketImportException {
+        Phenopacket phenopacket;
+        try (InputStream is = PhenopacketImportUtilTest.class.getResourceAsStream("pfeiffer.v2.pb")) {
+            phenopacket = PhenopacketImportUtil.readPhenopacket(is, Phenopacket.class);
+        }
+        assertThat(phenopacket.getId(), equalTo("pfeiffer-phenopacket"));
+        assertThat(phenopacket.getSubject().getId(), equalTo("II:3/Family 2"));
+    }
+
+}

